var/spack/repos/builtin/packages/sqlite/package.py

```diff
@@ -47,20 +48,25 @@ class Sqlite(AutotoolsPackage):
     # All versions prior to 3.26.0 are vulnerable to Magellan when FTS
     # is enabled, see https://blade.tencent.com/magellan/index_en.html
 
-    variant(
-        "functions",
-        default=False,
-        when="+dynamic_extensions",
-        description="Provide mathematical and string extension functions for SQL "
-        "queries using the loadable extensions mechanism",
-    )
-    variant("fts", default=True, description="Include fts4 and fts5 support")
-    variant("column_metadata", default=True, description="Build with COLUMN_METADATA")
-    variant("dynamic_extensions", default=True, description="Support loadable extensions")
-    variant("rtree", default=True, description="Build with Rtree module")
 
-    depends_on("readline")
+    # no hard readline dep on Windows + no variant support, makefile has minimal to no options
+    for plat in ["linux", "darwin", "cray"]:
```

Comment:
Needs `freebsd`.

Instead of these loops, let's get #41453 merged first.

Reply:
Oh geez, I apparently missed us adding `freebsd` somehow... thanks for pointing that out.

A. huge fan of #41453, thanks for adding that, these loops have been annoying me since day 1.
B. I'd prefer to let this land naturally and then go back and refactor after #41453 lands. #41453 hasn't received attention since the PR was made, this package is in immediate demand, and I don't want it to bit-rot here over the holidays. I'll have to refactor at least half a dozen packages after #41453 lands anyway, so this will just be one more.

Reply:
That was _very_ recent ;p two week ago?

Reply:
Ok, I guess I didn't really look at how long ago the PR was made, just that there were no recent updates. You're right, two weeks isn't that long, but given that your change is non trivial, and this is somewhat closer to that, I'd still prefer to not hold this up.

Reply:
I'm also a fan of merging this before https://github.com/spack/spack/pull/41453, but this should add `freebsd` to the list before it is merged.